contrib/guix/libexec/build.sh

```diff
@@ -196,19 +260,60 @@ export PATH="${BASEPREFIX}/${HOST}/native/bin:${PATH}"
             find "${DISTNAME}/lib" -type f -print0
         } | xargs -0 -n1 -P"$MAX_JOBS" -I{} "${DISTSRC}/contrib/devtools/split-debug.sh" {} {} {}.dbg
 
-        cp "${DISTSRC}/doc/README.md" "${DISTNAME}/"
+        case "$HOST" in
+            *mingw*)
+                cp "${DISTSRC}/doc/README_windows.txt" "${DISTNAME}/readme.txt"
+                ;;
+            *linux*)
+                cp "${DISTSRC}/doc/README.md" "${DISTNAME}/"
+                ;;
+        esac
 
         # Finally, deterministically produce {non-,}debug binary tarballs ready
         # for release
-        find "${DISTNAME}" -not -name "*.dbg" -print0 \
-            | sort --zero-terminated \
-            | tar --create --no-recursion --mode='u+rw,go+r-w,a+X' --null --files-from=- \
-            | gzip -9n > "${OUTDIR}/${DISTNAME}-${HOST}.tar.gz" \
-            || ( rm -f "${OUTDIR}/${DISTNAME}-${HOST}.tar.gz" && exit 1 )
-        find "${DISTNAME}" -name "*.dbg" -print0 \
-            | sort --zero-terminated \
-            | tar --create --no-recursion --mode='u+rw,go+r-w,a+X' --null --files-from=- \
-            | gzip -9n > "${OUTDIR}/${DISTNAME}-${HOST}-debug.tar.gz" \
-            || ( rm -f "${OUTDIR}/${DISTNAME}-${HOST}-debug.tar.gz" && exit 1 )
+        case "$HOST" in
+            *mingw*)
+                find "${DISTNAME}" -not -name "*.dbg" -print0 \
+                    | xargs -0r touch --no-dereference --date="@${SOURCE_DATE_EPOCH}"
+                find "${DISTNAME}" -not -name "*.dbg" \
+                    | sort \
```

Comment:
Can you explain the usage of `--zero-terminated` with `sort`? I see you're only using it in the linux case here, but are using it in the mingw case below. Is this something we can just pass all the time? If not, could this cause reproducibility issues if we forget it  in future?

I couldn't see it in the man page for `tar`, but I could find it for [`sed`](https://www.gnu.org/software/sed/manual/html_node/Command_002dLine-Options.html).

Reply:
Yes! The reason for this difference is that `tar` has a `--null` flag while `zip` does not.

The reason for using `-print0`, `--zero-terminated`, and `--null` is out of an abundance of caution: a newline character is a valid character in filenames, but `\0` is not. However, I guess this is defeated by the fact that `zip` doesn't have any options that support this. Perhaps a better thing is to error if `find` finds any filenames with newlines in the filename? Would like feedback.